packages/ckeditor5-dev-build-tools/tests/plugins/splitCss/splitCss.test.ts

```diff
@@ -0,0 +1,263 @@
+/**
+ * @license Copyright (c) 2003-2024, CKSource Holding sp. z o.o. All rights reserved.
+ * For licensing, see LICENSE.md.
+ */
+
+import { join } from 'path';
+import { test } from 'vitest';
+import { rollup, type RollupOutput, type NormalizedOutputOptions } from 'rollup';
+import { verifyDividedStyleSheet } from '../../_utils/utils.js';
+
+import { splitCss } from '../../../src/index.js';
+
+const TEST_BANNER = `/**
+ * License banner.
+ */
+`;
+
+/**
+ * Helper function for creating a bundle that won't be written to the file system.
+ */
+async function generateBundle(
+	input: string,
+	options?: NormalizedOutputOptions,
+	banner?: string
+): Promise<RollupOutput['output']> {
+	const bundle = await rollup( {
+		input,
+		plugins: [
+			splitCss()
+		]
+	} );
+
+	const { output } = await bundle.generate( { format: 'esm', banner } );
+
+	return output;
+}
+
+test( 'Import of single `CSS` file', async () => {
+	const output = await generateBundle( join( import.meta.dirname, './fixtures/single-import/input.ts' ) );
+	const expectedResult = `
+body {
+color: '#000';
+}
+`;
+
+	verifyDividedStyleSheet( output, 'styles.css', expectedResult );
+	verifyDividedStyleSheet( output, 'editor-styles.css', expectedResult );
+	verifyDividedStyleSheet( output, 'content-styles.css', '\n' );
+} );
+
+test( 'Import of single `CSS` file with a banner', async () => {
+	const output = await generateBundle( join( import.meta.dirname, './fixtures/single-import/input.ts' ), undefined, TEST_BANNER );
+	const expectedResult = `${ TEST_BANNER }
+body {
+color: '#000';
+}
+`;
+
+	const expectedEmptyResult = `${ TEST_BANNER }\n`;
+
+	verifyDividedStyleSheet( output, 'styles.css', expectedResult );
+	verifyDividedStyleSheet( output, 'editor-styles.css', expectedResult );
+	verifyDividedStyleSheet( output, 'content-styles.css', expectedEmptyResult );
+} );
+
+test( 'Import multiple `CSS` files', async () => {
+	const output = await generateBundle( join( import.meta.dirname, './fixtures/multiple-import/input.ts' ) );
+
+	const expectedResult = `
+body {
+color: '#000';
+}
+div {
+display: grid;
+}
+`;
```

Comment:
(Nitpick, can be ignored if you don't like it) You can create a test util that will access a string and remove all whitespaces after new line. This would all you to properly format the code in tests while still making the tests pass.

```ts
function removeWhitespace( text: string ): string {
	return text.replaceAll( /\n\s+/gm, '\n' );
}
```

Then you can call it like so:

const test2 = removeWhitespace( `
	body {
		color: '#000';
	}
	div {
		display: grid;
	}
` );